build.gradle

```diff
@@ -161,6 +157,110 @@ project ('hdfs') {
     }
 }
 
+project('test:integration') {
+    apply plugin: 'application'
+    applicationName = "pravega-selftest"
+    mainClassName = "io.pravega.test.integration.selftest.SelfTestRunner"
+
+    applicationDefaultJvmArgs = ["-server", "-Xmx4g", "-XX:+HeapDumpOnOutOfMemoryError",
+                                 "-Dlog.dir=PRAVEGA_APP_HOME/logs",
+                                 "-Dlog.name=selftest"]
+
+    applicationDistribution.from('src/config') {
+        into "conf"
+    }
+
+    test {
+                jvmArgs = ["-server", "-Xmx2g", "-XX:+HeapDumpOnOutOfMemoryError",
+                                 "-Dlog.dir=PRAVEGA_APP_HOME/logs",
+                                 "-Dlog.name=selftest"]
+    }
+
+    dependencies {
+        project(":hdfs")
+        project(":extendeds3")
+        compile "io.pravega:pravega-common:${pravegaVersion}",
+                "io.pravega:pravega-shared-metrics:${pravegaVersion}",
+                "io.pravega:pravega-shared-protocol:${pravegaVersion}",
+                "io.pravega:pravega-segmentstore-contracts:${pravegaVersion}",
+                "io.pravega:pravega-segmentstore-storage:${pravegaVersion}",
+                "io.pravega:pravega-segmentstore-server:${pravegaVersion}",
+                "io.pravega:pravega-segmentstore-server-host:${pravegaVersion}",
+                "io.pravega:pravega-controller:${pravegaVersion}",
+                "io.pravega:pravega-client:${pravegaVersion}"
+        compile  "io.pravega:pravega-test-testcommon:${pravegaVersion}"
+        compile group: 'junit', name:'junit', version: junitVersion
+        compile group: 'org.apache.curator', name: 'curator-test', version: apacheCuratorVersion
+        testCompile group: 'ch.qos.logback', name: 'logback-classic', version: qosLogbackVersion
+        testCompile group: 'org.apache.commons', name: 'commons-csv', version: apacheCommonsCsvVersion
+        compile group: 'org.apache.hadoop', name: 'hadoop-common', version: hadoopVersion, withoutLogger
+        compile group: 'org.apache.hadoop', name: 'hadoop-hdfs', version: hadoopVersion, withoutLogger
+        compile group: 'org.apache.hadoop', name: 'hadoop-hdfs-client', version: hadoopVersion, withoutLogger
+        compile group: 'org.apache.hadoop', name: 'hadoop-minicluster', version: hadoopVersion, withoutLogger
+        compile group: 'io.netty', name: 'netty-codec-http', version: nettyVersion
+        compile files(project(':hdfs').sourceSets.test.output)
+        compile group:"io.pravega", name:"pravega-segmentstore-server-host", version: pravegaVersion,  classifier: 'tests'
+        compile group:"io.pravega", name:"pravega-segmentstore-server", version: pravegaVersion,  classifier: 'tests'
+    }
+
+
+    task startServer(type: JavaExec) {
```

Comment:
please check whether all those tasks are actually needed in this repo. Looks like you just copy-pasted them from Core repository